### DISCUSSION TITLE
What is the advantage of incremental SSR over CDN or combining both?

### DISCUSSION THREAD

**Discussion by kartikag01** [vercel/next.js#17244] [Help]
### What is the advantage of incremental SSR over CDN or combining both?

If we implement `getStaticProps` with `getStaticPaths` with `fallback: true`  to generate pages via incremental SSR approach and using CDN above the nextjs to cache generate pages.

I have some questions about this architecture.

1. Should we remove CDN for caching HTML pages as incremental SSR is doing the same thing. if so does it increase/decrease TTFB? If we keep CDN then why we need incremental SSR?

2. When `revalidate` time passed, how to invalidate pages from CDN when we have combined both incremental SSR and CDN?

**Comment by leerob** [comment]
[Reposting from Reddit.](https://www.reddit.com/r/nextjs/comments/ix0ltu/what_is_the_advantage_of_incremental_ssr_over_cdn/g64dn26/?context=3)

`getStaticProps`/`getStaticPaths` are for static generation, *not* server-side rendering. When you use `fallback: true`, you are still serving a static fallback loading state while you make a request to the server for the new content. Saying SSR implies that rendering is blocked until the request from the server finishes.

The term "incremental SSR" isn't exactly correct. When using `getStaticProps`/`getStaticPaths` in combination with the `revalidate` option, here's what happens:

* You define a revalidation period of 1 second (e.g. `revalidate: 1`)
* At build time, your pages are statically generated
* A request is made to your page and the static file is served
* Three seconds later, another request is made and a static page is served. The revalidation period has passed, so a request to fetch the newest content from the server is made in the background.
* If more traffic hits during this time, it will get the currently cached static page.
* Once the new data is fetched from the server, the cache is invalidated and the new static page is now served.
* Another user visits your page and sees the most updated content.

If you're deploying with Vercel, they make the CDN / caching easy. You can use `revalidate` \+ Next.js without Vercel, but it's nice to not have to mess with caching.
